compiler/lambdalifting.nim

```diff
@@ -417,6 +418,9 @@ Consider:
 
 """
 
+proc isTypeOf(ident: PIdent; cache: IdentCache): bool =
+  ident == getIdent(cache, "typeof") or ident == getIdent(cache, "type")
+
```

Comment:
Why does the frontend not transform these into `nkType` nodes? It should and would make this far easier.

Reply:
I was expecting these are just `nkTypeOfExpr`

Reply:
Yes, even better. :-)

Reply:
what's the node processing path goes to here?

Reply:
`semMagicAfterOverloadResolution` should turn the magic mType and mTypeOf to nkTypeOfExpr

Reply:
tried , didn't found chance transform the node , now only detect via magic, seems sufficient